src/LoaderFactory.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Part of the Joomla Framework Language Package
+ *
+ * @copyright  Copyright (C) 2005 - 2015 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE
+ */
+
+namespace Joomla\Language;
+
+/**
+ * Factory for creating file loaders
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class LoaderFactory
+{
+	/**
+	 * A map of the registered loaders
+	 *
+	 * @var    LoaderInterface[]
+	 * @since  __DEPLOY_VERSION__
+	 */
+	private $loaderMap = [];
+
+	/**
+	 * Add a loader for a file type
+	 *
+	 * @param   string           $type    The loader type to register (typically the file extension)
+	 * @param   LoaderInterface  $loader  The loader to use for the type
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function addLoader(string $type, LoaderInterface $loader)
```

Comment:
Rather than this I'd have a `getType` method in the `LoaderInterface`

Reply:
Changed.